src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(Map<String, Integer> inventory, FruitTransaction fruitTransaction) {
+        inventory.put(fruitTransaction.getFruit(),
+                 inventory.getOrDefault(fruitTransaction
+                        .getFruit(), 0)
+                - fruitTransaction.getQuantity());
+    }
```

Comment:
According to the checklist, the PurchaseOperation should check if the result balance is positive before saving it in the Storage. It's important to ensure that you are not allowing the inventory to go into negative quantities, which would be illogical for a real-world inventory system. You should throw a RuntimeException if the balance after the purchase would be negative.